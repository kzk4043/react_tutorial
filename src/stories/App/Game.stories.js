import Game from "../../App";
import { within, userEvent } from "@storybook/testing-library";
import { expect } from "@storybook/jest";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: "Tic-Tac-Toe/Game",
  component: Game,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
// Gameの初期状態
export const Initial = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const squares = canvas.getAllByRole("button");
    await userEvent.click(squares[0]);
    await userEvent.click(squares[1]);
    await userEvent.click(squares[2]);
    await userEvent.click(squares[3]);
    await userEvent.click(squares[4]);
    await userEvent.click(squares[5]);
    await userEvent.click(squares[6]);

    // 👇 Assert DOM structure
    await expect(canvas.getByText("Winner: X")).toBeInTheDocument();
  },
};

// 引き分けの時は「Draw!」と表示される
// 引き分けなので、勝利メッセージが表示されないことを確認
export const Draw = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const squares = canvas.getAllByRole("button");
    await userEvent.click(squares[0]);
    await userEvent.click(squares[1]);
    await userEvent.click(squares[2]);
    await userEvent.click(squares[3]);
    await userEvent.click(squares[5]);
    await userEvent.click(squares[4]);
    await userEvent.click(squares[6]);
    await userEvent.click(squares[8]);
    await userEvent.click(squares[7]);

    // 👇 Assert DOM structure
    await expect(canvas.getByText("Draw!")).toBeInTheDocument();
    await expect(canvas.queryByText("Winner")).not.toBeInTheDocument();
  },
};
