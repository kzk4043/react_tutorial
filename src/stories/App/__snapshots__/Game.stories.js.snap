// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tic-Tac-Toe/Game Initial play-test 1`] = `
<div class="game">
  <div class="game-board">
    <div class="status">
      Winner: X
    </div>
    <div class="board-row">
      <button class="square">
        X
      </button>
      <button class="square">
        O
      </button>
      <button class="square">
        X
      </button>
    </div>
    <div class="board-row">
      <button class="square">
        O
      </button>
      <button class="square">
        X
      </button>
      <button class="square">
        O
      </button>
    </div>
    <div class="board-row">
      <button class="square">
        X
      </button>
      <button class="square">
      </button>
      <button class="square">
      </button>
    </div>
  </div>
  <div class="game-info">
    <ol>
      <li>
        <button>
          Go to game start
        </button>
      </li>
      <li>
        <button>
          Go to move #1
        </button>
      </li>
      <li>
        <button>
          Go to move #2
        </button>
      </li>
      <li>
        <button>
          Go to move #3
        </button>
      </li>
      <li>
        <button>
          Go to move #4
        </button>
      </li>
      <li>
        <button>
          Go to move #5
        </button>
      </li>
      <li>
        <button>
          Go to move #6
        </button>
      </li>
      <li>
        <button>
          Go to move #7
        </button>
      </li>
    </ol>
  </div>
</div>
`;
